csharp/DEVELOPER.md

```diff
@@ -4,30 +4,68 @@ This document describes how to set up your development environment to build and
 
 ### Development Overview
 
-The GLIDE C# wrapper consists of both C# and Rust code.
+We're excited to share that the GLIDE C# client is currently in development! However, it's important to note that this client is a work in progress and is not yet complete or fully tested. Your contributions and feedback are highly encouraged as we work towards refining and improving this implementation. Thank you for your interest and understanding as we continue to develop this C# wrapper.
+
+The C# client contains the following parts:
+
+1. Rust part of the C# client located in `lib/src`; it communicates with [GLIDE core rust library](../glide-core/README.md).
+2. C# part of the client located in `lib`; it translates Rust async API into .Net async API.
+3. Integration tests for the C# client located in `tests` directory.
+4. A dedicated benchmarking tool designed to evaluate and compare the performance of GLIDE for Redis and other .Net clients. It is located in `<repo root>/benchmarks/csharp`.
+
+TODO: examples, UT, design docs
 
 ### Build from source
 
+Software Dependencies:
+
+- .Net SDK 6 or later
+- git
+- rustup
+
+Please also install the following packages to build [GLIDE core rust library](../glide-core/README.md):
+
+- GCC
+- protoc (protobuf compiler)
+- pkg-config
+- openssl
+- openssl-dev
+
 #### Prerequisites
 
-Software Dependencies
+**.Net**
 
--   .net sdk 6 or later
--   git
--   GCC
--   pkg-config
--   protoc (protobuf compiler)
--   openssl
--   openssl-dev
--   rustup
+It is recommended to visit https://dotnet.microsoft.com/en-us/download/dotnet to download .Net installer.
+You can also use a package manager to install the .Net SDK:
+
+```bash
+brew install dotnet@6         # MacOS
+sudo apt-get install dotnet6  # Linux
+```
+
+**Protoc installation**
+
+Download a binary matching your system from the [official release page](https://github.com/protocolbuffers/protobuf/releases/tag/v25.1) and make it accessible in your $PATH by moving it or creating a symlink.
+For example, on Linux you can copy it to `/usr/bin`:
+
+```bash
+sudo cp protoc /usr/bin/
+```
+
+**Dependencies installation for Ubuntu**
+
+```bash
+sudo apt-get update -y
+sudo apt-get install -y openssl openssl-dev gcc
+curl --proto '=https' --tlsv1.2 -sSf https://sh.rustup.rs | sh
+source "$HOME/.cargo/env"
+```
 
 **Dependencies installation for MacOS**
 
-visit https://dotnet.microsoft.com/en-us/download/dotnet
-to download .net installer
 ```bash
 brew update
-brew install git gcc pkgconfig protobuf openssl
```

Comment:
isn't protoc invoked during the rust build process?

Reply:
We need a specific version[^1], so I added a block describing how to install it (lines 46-49).

[^1]: At least for java, but I prefer to keep instructions aligned